src/controllers/StaffController.js

```diff
@@ -553,6 +625,580 @@ class StaffController {
+    const assignedData = await staff.assignCourses(id, data);
+    if (assignedData.error) {
+      return res.status(400).json({
+        error: assignedData.error,
+      });
+    }
+    const operatorData = await staff.getDashboardData();
+    if (!operatorData) {
+      return res.status(400).json({ error: 'Internal Server Error fetching Dashboard' });
+    }
+    return res.status(200).json({
+      operation: 'Course assigned successfully',
+      xToken: token,
+      assignedData,
+      operatorData,
+    });
+  }
+
+  /**
+   * Approves an object in the specified collection for admin access.
+   * @async
+   * @static
+   * @param {Object} req - The request object.
+   * @param {Object} res - The response object.
+   * @returns {Promise<Object>} The result of approving an object.
+   * @throws {Error} If there is an error during the approval process.
+   */
+  static async adminApproveRecord(req, res) {
+    // Extract the ID from the request parameters
+    const { id } = req.params;
+    if (!id) {
+      return res.status(400).json({ error: 'Missing ID' });
+    }
+    const adminFxn = await authClient.staffPreCheck(req);
+    if (adminFxn.error) {
+      return res.status(401).json({
+        error: adminFxn.error,
+        msg: 'Staff is not authorized to perform this action',
+      });
+    }
+    const { staff, token } = adminFxn;
+    const recordUpdate = await staff.approveRecord(id);
+    if (recordUpdate.error) {
+      return res.status(400).json({
+        error: recordUpdate.error,
+      });
+    }
+    const operatorData = await staff.getDashboardData();
+    if (!operatorData) {
+      return res.status(400).json({ error: 'Internal Server Error fetching Dashboard' });
+    }
+    return res.status(200).json({
+      operation: 'Record approved successfully',
+      xToken: token,
+      recordUpdate,
+      operatorData,
+    });
+  }
+
+  /**
+   * Fetches available courses for the specified collection.
+   * @async
+   * @static
+   * @param {Object} req - The request object.
+   * @param {Object} res - The response object.
+   * @returns {Promise<Object>} The result of fetching available courses.
+   * @throws {Error} If there is an error during the fetch process.
+   */
+  static async adminGetAvailableCourses(req, res) {
+    // Extract the ID from the request parameters
+    const { id } = req.params;
+    if (!id) {
+      return res.status(400).json({ error: 'Missing ID' });
+    }
+    const data = req.body;
+    if (!data) {
+      return res.status(400).json({ error: 'Missing data for the request' });
+    }
+    const typeCollection = data.collection;
+    if (!typeCollection) {
+      return res.status(400).json({
+        error: 'Missing collection type for the request',
+        msg: 'Ensure the key is defined as \'collection\'',
+      });
+    }
+    const validCollectionTypes = ['Department', 'Faculty'];
+    if (!validCollectionTypes.includes(typeCollection)) {
```

Comment:
_Same as above. Code duplication: Called method already performs this check. . ._